### PR TITLE
Admin Generator (Future): Implement static-select type for combination column

### PR DESCRIPTION
**Pull request by jamesricky**
---

## PR Checklist

-   [x] Verify if the change requires a changeset. See [CONTRIBUTING.md](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md)
-   [x] Link to the respective task if one exists: SVK-367

### CODE REVIEW

File: demo/admin/src/products/future/generated/CombinationFieldsTestProductsGrid.tsx [hunk 1]
```diff
@@ -116,6 +118,48 @@ export function ProductsGrid({ toolbarAction, rowAction }: Props): React.ReactEl
...
         },
+        {
+            field: "staticSelectType",
+            headerName: intl.formatMessage({ id: "product.staticSelectType", defaultMessage: "Type (static select)" }),
+            filterable: false,
+            sortable: false,
+            renderCell: ({ row }) => {
+                const primaryEmptyMessage = "-";
+                const typePrimaryLabels: Record<string, string> = {
+                    Cap: intl.formatMessage({ id: "product.staticSelectType.primaryText.Cap", defaultMessage: "This is a cap" }),
```
Comment (johnnyomair) [inline demo/admin/src/products/future/generated/CombinationFieldsTestProductsGrid.tsx:129]: I'd prefer FormattedMessage here, is that possible?

File: packages/admin/cms-admin/src/generator/future/utils/stringManipulation.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export const firstCharUpperCase = (str: string) => {
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/generator/future/utils/stringManipulation.ts:1]: Please use the change-case package here, which we already use for [camelCaseToHumanReadable](https://github.com/vivid-planet/comet/blob/main/packages/admin/cms-admin/src/generator/future/utils/camelCaseToHumanReadable.ts)

**Comment by johnnyomair** [comment]
General question: Do we plan to add support for using multiple different types (text, select, number etc.) in combination? Currently it looks like you're only supporting one at a time.

**Comment by jamesricky** [comment]
> General question: Do we plan to add support for using multiple different types (text, select, number etc.) in combination? Currently it looks like you're only supporting one at a time.

Yes, working on that 🙂

File: packages/admin/cms-admin/src/generator/future/utils/stringManipulation.ts (hunk 2 above)
Comment (jamesricky) [reply packages/admin/cms-admin/src/generator/future/utils/stringManipulation.ts]: Cool 👍🏼 
Done in fc799fba26bec4579cabf3f723a34f65cb85075b

File: demo/admin/src/products/future/generated/CombinationFieldsTestProductsGrid.tsx (hunk 1 above)
Comment (jamesricky) [reply demo/admin/src/products/future/generated/CombinationFieldsTestProductsGrid.tsx]: Yes: bcbe408fe785d2215a3add2df5ed387d92a1164e

**Review by johnnyomair — DISMISSED** [review]

**Review by nsams — APPROVED** [review]

**Comment by johnnyomair** [comment]
@jamesricky please resolve conflicts (main merge), then this should be good to go.

**Comment by johnnyomair** [comment]
@jamesricky please resolve conflicts

File: demo/admin/src/products/future/generated/CombinationFieldsTestProductsGrid.tsx [hunk 3]
```diff
@@ -148,7 +148,7 @@ export function ProductsGrid({ toolbarAction, rowAction }: Props): React.ReactEl
             renderCell: ({ row }) => {
                 const primaryEmptyMessage = <FormattedMessage id="product.staticSelectInStock.primaryText.empty" defaultMessage="No stock info" />;
                 const inStockPrimaryLabels: Record<string, React.ReactNode> = {
-                    true: <FormattedMessage id="product.staticSelectInStock.primaryText.true" defaultMessage="It's in stock :D" />,
+                    true: <FormattedMessage id="product.staticSelectInStock.primaryText.true" defaultMessage={`It's in stock :D`} />,
```
Comment (johnnyomair) [inline demo/admin/src/products/future/generated/CombinationFieldsTestProductsGrid.tsx:151]: Why is this necessary?

Comment (jamesricky) [reply demo/admin/src/products/future/generated/CombinationFieldsTestProductsGrid.tsx:151]: In case the given string includes double-quotes e.g., here:

https://github.com/vivid-planet/comet/blob/0526d525c62b17eba6fdc0b5ed7614d6c2b59065/demo/admin/src/products/future/generated/CombinationFieldsTestProductsGrid.tsx#L258

Usually prettier replaces the `{``}` with double quotes if they are not necessary, however since this string includes a single quote it is not replaced. 

Should I add a check to only do the `{``}` syntax if the string includes a double quote?

**Review by johnnyomair — APPROVED** [review]

**Review by nsams — APPROVED** [review]
